Bmx.Idp.Okta/OktaClient.cs

```diff
@@ -13,32 +8,26 @@
 
 namespace Bmx.Idp.Okta {
 	public class OktaClient : IIdentityProvider {
-		private readonly JsonSerializerOptions _serializeOptions =
-			new JsonSerializerOptions {PropertyNamingPolicy = JsonNamingPolicy.CamelCase,};
-
-		private readonly CookieContainer _cookieContainer;
-		private readonly HttpClient _httpClient;
-
+		private readonly IOktaApi _oktaApi;
 		private string _oktaStateToken;
 		private string _oktaSessionToken;
 		private OktaMfaFactor[] _oktaMfaFactors;
 		private OktaApp[] _oktaApps;
 
-		public OktaClient() {
-			_cookieContainer = new CookieContainer();
-			_httpClient = new HttpClient( new HttpClientHandler {CookieContainer = _cookieContainer} );
-			_httpClient.Timeout = TimeSpan.FromSeconds( 30 );
-			_httpClient.DefaultRequestHeaders.Accept.Add( new MediaTypeWithQualityHeaderValue( "application/json" ) );
+		public OktaClient( IOktaApi oktaApi = null ) {
+			if( oktaApi == null ) {
+				_oktaApi = new OktaApi();
```

Comment:
Not sure how I feel about this pattern... Normally I'd just always inject the dependency.